errors.go

```diff
@@ -151,6 +152,24 @@ func GetUserMessage(err error) string {
 	return ""
 }
 
+// detailedError identifies an error with details
+type detailedError interface {
+	Details() []interface{}
+}
+
+// Details returns the error details
+func (c *customError) Details() []interface{} { return c.details }
+
+// GetDetails returns a slice of arbitratry details for all errors
+// If error is not `detailedError` returns nil
+func GetDetails(err error) []interface{} {
+	if detailedError, ok := err.(detailedError); ok {
+		return detailedError.Details()
+	}
+
+	return nil
```

Comment:
I am not sure if it's better to return `nil` or empty slice.
I think both can be tested for as `if len(details) == 0` so we can leave this as nil.
This discussion continues on line 253 with the usage of this function

@cben @tzvatot WDYT?

Reply:
either one will be captured by the json encoder `omitempty`, in which case it doesn't matter much. Returning an empty slice simplifies the code further down.